app/controllers/launches_controller.rb

```diff
@@ -55,6 +55,11 @@ def create
       find_or_create_user
       find_or_create_enrollment
       set_current_enrollment
+      @submission = Submission.find_or_create_by(
+        resource: @resource,
+        enrollment: @enrollment,
+      )
+      set_current_submission
```

Comment:
This is kind of a nit-pick, but if you've already made helper methods somewhere like `find_or_create_user` and `find_or_create_enrollment`, it might make sense to also do that for the `Submission`.